### PR TITLE
fix(public_gateway): added retry logic for public gateway deletion

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_public_gateway.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	"fmt"
 	"log"
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_public_gateway.go:9]: Can you please add your terraform test template and the result in the PR.

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Hi Ujjwal, 
Can you please add your terraform test template and the result in the PR.

**Comment by SunithaGudisagarIBM** [comment]
```
resource "ibm_is_vpc" "testacc_vpc" {
  name = "sunitha"
}
resource "ibm_is_public_gateway" "testacc_public_gateway" {
  name           = "sunitha-gateway"
  vpc            = "r006-567a5aec-d475-42cd-9407-052b1d4c167d"
  # vpc         = ibm_is_vpc.testacc_vpc.id
  zone           = "us-south-1"
  # resource_group = "7494aacb866142fba11a88d75cb37bd8"
}

resource "ibm_is_subnet" "testacc_subnet" {
  name            = "sunitha-subnet"
  vpc             = ibm_is_vpc.testacc_vpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
  public_gateway  = ibm_is_public_gateway.testacc_public_gateway.id
  # tags            = ["tag1"]
}
```
Receiving the below error :
Screenshot 2023-01-24 at 2 07 08 PM

**Comment by SunithaGudisagarIBM** [comment]
Issue is not yet solved, can you please check this scenario
```
resource "ibm_is_vpc" "testacc_vpc" {
  name = "sunitha"
}
# resource "ibm_is_public_gateway" "testacc_public_gateway" {
#   name           = "sunitha-gateway"
# #   vpc            = "r006-567a5aec-d475-42cd-9407-052b1d4c167d"
#   vpc         = ibm_is_vpc.testacc_vpc.id
#   zone           = "us-south-1"
#   # resource_group = "7494aacb866142fba11a88d75cb37bd8"
# }

resource "ibm_is_subnet" "testacc_subnet" {
  name            = "sunitha-subnet"
  vpc             = ibm_is_vpc.testacc_vpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
#   public_gateway  = ibm_is_public_gateway.testacc_public_gateway.id
  # tags            = ["tag1"]
}
```
Screenshot 2023-02-06 at 6 14 33 PM

**Comment by ujjwal-ibm** [comment]
working fine now 

```
Plan: 1 to add, 1 to change, 1 to destroy.
ibm_is_public_gateway.gateway: Destroying... [id=r006-87da9566-cb8c-45fa-93b1-ef0c3d131e5f]
ibm_is_public_gateway.gateway: Still destroying... [id=r006-87da9566-cb8c-45fa-93b1-ef0c3d131e5f, 10s elapsed]
ibm_is_public_gateway.gateway: Still destroying... [id=r006-87da9566-cb8c-45fa-93b1-ef0c3d131e5f, 20s elapsed]
ibm_is_public_gateway.gateway: Destruction complete after 26s
ibm_is_public_gateway.gateway: Creating...
ibm_is_public_gateway.gateway: Still creating... [10s elapsed]
ibm_is_public_gateway.gateway: Creation complete after 14s [id=r006-575909e1-c25e-498a-9cf1-676417a5125d]
ibm_is_subnet.subnet: Modifying... [id=0717-02648246-c3b2-4b86-a884-5aa1f6928c3b]
ibm_is_subnet.subnet: Still modifying... [id=0717-02648246-c3b2-4b86-a884-5aa1f6928c3b, 10s elapsed]
ibm_is_subnet.subnet: Modifications complete after 13s [id=0717-02648246-c3b2-4b86-a884-5aa1f6928c3b]

Apply complete! Resources: 1 added, 1 changed, 1 destroyed.
```

**Comment by ujjwal-ibm** [comment]
```
=== RUN   TestAccIBMISPublicGateway_resource_group_change
--- PASS: TestAccIBMISPublicGateway_resource_group_change (154.60s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     157.011s
```

**Comment by ujjwal-ibm** [comment]
moving to upstream

**Pull request closed without merging**